src/confluent_kafka/aio/_AIOProducer.py

```
@@ -60,47 +142,254 @@ async def _call(self, blocking_task, *args, **kwargs):
+                for j, msg_data in enumerate(group_data['messages']):
+                    self._message_buffer.append(msg_data)
+                    self._buffer_futures.append(group_data['futures'][j])
+                continue
+            
+            # Convert messages to batch format
+            batch_messages = []
+            for msg_data in group_data['messages']:
+                batch_msg = {
+                    'value': msg_data['value'],
+                    'key': msg_data['key']
+                }
+                
+                # Add optional fields if present
+                if 'partition' in msg_data:
+                    batch_msg['partition'] = msg_data['partition']
+                if 'timestamp' in msg_data:
+                    batch_msg['timestamp'] = msg_data['timestamp']
+                if 'headers' in msg_data:
+                    batch_msg['headers'] = msg_data['headers']
+                
+                batch_messages.append(batch_msg)
+            
+            # Setup callback with pre-allocated state
+            batch_futures = group_data['futures']
+            batch_user_callbacks = group_data['callbacks']
+            callback_state = {'index': 0, 'futures': batch_futures, 'callbacks': batch_user_callbacks}
+            
+            def batch_delivery_callback(err, msg):
+                idx = callback_state['index']
+                if idx < len(callback_state['futures']):
+                    future = callback_state['futures'][idx]
+                    user_callback = callback_state['callbacks'][idx]
+                    
+                    if err:
+                        future.set_exception(_KafkaException(err))
+                    else:
+                        future.set_result(msg)
+                        if user_callback:
+                            try:
+                                user_callback(err, msg)
+                            except Exception:
```

I would not try to catch all exceptions here, let them file and debug themselves. If they want exception swallowing let the user_callback define it inside their function.